### PR TITLE
Reduce double and float precision requirements on rest CSV tests

### PR DESCRIPTION
**Pull request by ivancea**
Fixes https://github.com/elastic/elasticsearch-serverless/issues/2837

The failing value is `5.801464200000001`, which rounds to `5.8014642`. However, `5.8014642` is roudned to `5.801464199`.
With a precision of 7, both are truncated to `5.801464`.

Not the most elegant solution, but it works for this case, which may be a quite edgy one.

### CODE REVIEW

File: x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvAssert.java [hunk 1]
```diff
@@ -337,6 +346,42 @@ private static Object rebuildExpected(Object expectedValue, Class<?> clazz, Func
         }
     }
 
+    /**
+     * Creates an equality checker that handles both single elements and lists of elements.
+     */
+    @SuppressWarnings("unchecked")
+    private static <T> BiFunction<Object, Object, Boolean> equalityChecker(
```
Comment (ivancea) [inline x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvAssert.java:353]: Extracted this to a generic method, as we may need it for other types, like GEO types (E.g. centroid precision error?)

**Comment by ivancea** [comment]
After seeing this: https://github.com/elastic/elasticsearch/pull/110238

I'll reconsider it. It would be weird if both solutions co-exist

**Comment by ivancea** [comment]
Reopened with a different approach for doubles, and discarded the old one, as non-rest CSV tests apparently have no issues with doubles

**Comment by ivancea** [comment]
@elasticmachine update branch

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @ivancea.

**Pull request merged**